### PR TITLE
[CMDCT-3145] Report period and year fix

### PR DESCRIPTION
**Pull request by ntsummers1**
### Description

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/03f8681f-7f42-44d1-b7d9-65edf14ea622

### Related ticket(s)

CMDCT-3145

---
### How to test

Make sure that your isCopyOverTest is set to true on line 44 in AddEditReportModal `const isCopyOverTest = true`
Create a work plan and submit it as a state user
Approve that work plan as an admin user
Copy over the work plan as a state user
See that its been copied into the new year!

### Important updates

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [x] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/app-api/utils/constants/constants.ts
Comment (cassandradanger) [inline services/app-api/utils/constants/constants.ts:20]: maybe we want to pass this by product? My only update would be
"Unable to copy over report if todays date is `in` the same period and year as a previous report."

Comment (cassandradanger) [inline services/app-api/utils/constants/constants.ts:20]: maybe we want to pass this by product? My only update would be
"Unable to copy over report if todays date is in the same period and year as a previous report."

**Review by cassandradanger — DISMISSED** [review]
i think this looks good!! thanks for the quick turnaround!
one little nit - but not holding up approval for it - will re-approve if needed :)

File: services/app-api/utils/constants/constants.ts
Comment (ntsummers1) [reply services/app-api/utils/constants/constants.ts:20]: This is a message that only gets sent back as an api error to help us debug, so its really only help to us for debugging purposes. That said, I can make the change!

Comment (ntsummers1) [inline services/app-api/utils/constants/constants.ts:20]:
```suggestion
    "Unable to copy over report if todays date is in the same period and year as a previous report.",
```

File: services/app-api/handlers/reports/create.ts
Comment (britt-mo) [inline services/app-api/handlers/reports/create.ts:147]: I'm having trouble understanding line 147.  If it's an override copy over, and the report period is `1`, then it should be `2`, otherwise it'll be set to `1`?

**Review by gmrabian — COMMENTED** [review]
Code changes look reasonable to me. Could you explain what was happening with dev and val and why the year was incorrectly incremented in val? I'm also curious how copy over plays into that, because I tested it by archiving existing reports and creating a new, non-copied one

**Review by gmrabian — APPROVED** [review]

**Review by britt-mo — APPROVED** [review]

**Pull request merged**
